src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -6,18 +6,32 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        // Order of these properties matter because GetUseIcuMode is dependent on Invariant.
+        // So we need Invariant to be initialized first.
         internal static bool Invariant { get; } = GetGlobalizationInvariantMode();
 
-        // GetInvariantSwitchValue calls CLRConfig first to detect if the switch is defined in the config file.
+#if TARGET_WINDOWS
+        internal static bool UseIcu { get; } = GetUseIcuMode();
+#else
+        // We can't just call GetUseIcuMode and return false there if we want the linker to trim out the Windows code.
+        internal static bool UseIcu => true;
+#endif
+
+        internal static bool GetInvariantSwitchValue() =>
+            GetSwitchValue("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
+
+        internal static bool GetUseNlsSwitchValue() =>
```

Comment:
This can be in the .Windows.cs file too.

Reply:
Make sense.

Reply:
And maybe used directly, without the wrapper.